lwip/esp_interface.c

```diff
@@ -152,30 +152,51 @@ void pp_recycle_rx_pbuf(struct pbuf *p)
     LWIP_ASSERT("expected esf_buf", p->esf_buf);
     sdk_system_pp_recycle_rx_pkt(p->esf_buf);
     taskENTER_CRITICAL();
+    LWIP_ASSERT("pp_rx_pool_usage underflow", pp_rx_pool_usage > 0);
     pp_rx_pool_usage--;
     taskEXIT_CRITICAL();
 }
 
+/* Set to true to copy the rx pbufs on input and free them. The pp rx buffer
+ * pool is limited so this allows a large number at the expense of memory.
+ */
+
+#define COPY_PP_RX_PBUFS 0
+
 /* Return the number of ooseq bytes that can be retained given the current
  * size 'n'. */
 size_t ooseq_max_bytes(size_t n)
 {
+#if COPY_PP_RX_PBUFS
     size_t free = xPortGetFreeHeapSize();
-    size_t target = (free - 30000) + n;
+    ssize_t target = ((ssize_t)free - 30000) + n;
```

Comment:
Out of curiosity, is 30000 just an approximate arbitrary number or is it tied to an allocation somewhere?